File: frontend/src/api/testSuite.ts
import request from '@/utils/request'

// ==================== 类型定义 ====================

export interface TestSuiteCreate {
  project_id: number
  name: string
  description?: string
  case_ids: number[]
  environment_id?: number
  concurrency?: 1 | 3 | 5 | 10
  failure_strategy?: 'continue' | 'stop'
  variables?: Record<string, string>
  tags?: string[]
}

export interface TestSuiteUpdate {
  name?: string
  description?: string
  case_ids?: number[]
  environment_id?: number
  concurrency?: 1 | 3 | 5 | 10
  failure_strategy?: 'continue' | 'stop'
  variables?: Record<string, string>
  tags?: string[]
}

export interface TestSuiteInfo {
  id: number
  project_id: number
  name: string
  description?: string
  case_ids: number[]
  case_count: number
  environment_id?: number
  environment_name?: string
  concurrency: number
  failure_strategy: string
  variables: Record<string, string>
  tags: string[]
  creator_id?: number
  created_at: string
  updated_at?: string
}

export interface TestSuiteListItem {
  id: number
  project_id: number
  project_name?: string
  name: string
  description?: string
  case_count: number
  environment_name?: string
  concurrency: number
  failure_strategy: string
  tags: string[]
  creator_id?: number
  created_at: string
}

export interface TestSuiteRunRequest {
  environment_id?: number
  concurrency?: 1 | 3 | 5 | 10
  failure_strategy?: 'continue' | 'stop'
  variables?: Record<string, string>
}

// ==================== API 函数 ====================

export const createTestSuite = (data: TestSuiteCreate) => {
  return request<TestSuiteInfo>({
    url: '/test-suites',
    method: 'post',
    data
  })
}

export const getTestSuites = (params?: {
  project_id?: number
  keyword?: string
  tag?: string
  page?: number
  page_size?: number
}) => {
  return request<{ total: number; page: number; page_size: number; items: TestSuiteListItem[] }>({
    url: '/test-suites',
    method: 'get',
    params
  })
}

export const getTestSuite = (id: number) => {
  return request<TestSuiteInfo>({
    url: `/test-suites/${id}`,
    method: 'get'
  })
}

export const updateTestSuite = (id: number, data: TestSuiteUpdate) => {
  return request<TestSuiteInfo>({
    url: `/test-suites/${id}`,
    method: 'put',
    data
  })
}

export const deleteTestSuite = (id: number) => {
  return request({
    url: `/test-suites/${id}`,
    method: 'delete'
  })
}

export const copyTestSuite = (id: number) => {
  return request<TestSuiteInfo>({
    url: `/test-suites/${id}/copy`,
    method: 'post'
  })
}

export const runTestSuite = (id: number, data?: TestSuiteRunRequest) => {
  return request<{ id: number; message: string }>({
    url: `/test-suites/${id}/run`,
    method: 'post',
    data
  })
}

File: server/schemas/test_suite.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class TestSuiteCreate(BaseModel):
    """创建任务配置"""
    project_id: int
    name: str
    description: Optional[str] = None
    config_mode: str = "simple"  # simple / orchestration
    case_ids: list[int] = []
    environment_id: Optional[int] = None
    concurrency: int = 1
    failure_strategy: str = "continue"
    variables: dict[str, str] = {}
    tags: list[str] = []


class TestSuiteUpdate(BaseModel):
    """更新任务配置"""
    name: Optional[str] = None
    description: Optional[str] = None
    config_mode: Optional[str] = None
    case_ids: Optional[list[int]] = None
    environment_id: Optional[int] = None
    concurrency: Optional[int] = None
    failure_strategy: Optional[str] = None
    variables: Optional[dict[str, str]] = None
    tags: Optional[list[str]] = None


class TestSuiteInfo(BaseModel):
    """任务配置详情"""
    id: int
    project_id: int
    name: str
    description: Optional[str] = None
    config_mode: str = "simple"
    case_ids: list[int] = []
    case_count: int = 0  # 用例数量
    environment_id: Optional[int] = None
    environment_name: Optional[str] = None
    concurrency: int = 1
    failure_strategy: str = "continue"
    variables: dict[str, str] = {}
    tags: list[str] = []
    creator_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class TestSuiteList(BaseModel):
    """任务配置列表项"""
    id: int
    project_id: int
    project_name: Optional[str] = None
    name: str
    description: Optional[str] = None
    config_mode: str = "simple"
    case_count: int = 0
    environment_name: Optional[str] = None
    concurrency: int = 1
    failure_strategy: str = "continue"
    tags: list[str] = []
    creator_id: Optional[int] = None
    created_at: datetime

    class Config:
        from_attributes = True


class TestSuiteRunRequest(BaseModel):
    """执行任务配置请求（可覆盖默认配置）"""
    environment_id: Optional[int] = None
    concurrency: Optional[int] = None
    failure_strategy: Optional[str] = None
    variables: Optional[dict[str, str]] = None
